library/std/math.qs

```diff
@@ -499,7 +665,288 @@ namespace Microsoft.Quantum.Math {
             set size = size + 1;
             set number = number >>> 1;
         }
-        return size;
+
+        size
+    }
+
+    /// # Summary
+    /// For a non-negative integer `a`, returns the number of bits required to represent `a`.
+    /// NOTE: This function returns the smallest n such that a < 2^n.
+    function BitSizeL(a : BigInt) : Int {
+        Fact(a >= 0L, "`a` must be non-negative.");
+        mutable number = a;
+        mutable size = 0;
+        while (number != 0L) {
+            set size = size + 1;
+            set number = number >>> 1;
+        }
+
+        size
+    }
+
+    //
+    // Combinatorics
+    //
+
+    /// # Summary
+    /// Returns the factorial of a given number.
+    ///
+    /// # Description
+    /// Returns the factorial of a given nonnegative integer n, where 0 ≤ n ≤ 20.
+    ///
+    /// # Input
+    /// ## n
+    /// The number to take the factorial of.
+    ///
+    /// # Output
+    /// The factorial of `n`.
+    ///
+    /// # Remarks
+    /// For inputs greater than 20, please use `Microsoft.Quantum.Math.FactorialL`.
+    ///
+    /// # See Also
+    /// - Microsoft.Quantum.Math.FactorialL
+    /// - Microsoft.Quantum.Math.ApproximateFactorial
+    function FactorialI(n : Int) : Int {
+        Fact(n >= 0, "The factorial is not defined for negative inputs.");
+        Fact(n <= 20, "The largest factorial that be stored as an Int is 20!. Use FactorialL or ApproximateFactorial.");
+
+        mutable result = 1;
+        for i in 1 .. n {
+            set result *= i;
+        }
+        result
+    }
+
+    /// # Summary
+    /// Returns the factorial of a given number.
+    ///
+    /// # Description
+    /// Returns the factorial of a given nonnegative integer n, where 0 ≤ n ≤ 20.
```

Comment:
Cut & paste error? I thought the L version was good for n > 20.

Reply:
Cut & paste error indeed.